.github/workflows/on_pr.yaml

```diff
@@ -0,0 +1,55 @@
+name: Test GlassFlow Python SDK
+
+on:
+  pull_request:
+    branches:
+      - main
+
+jobs:
+  tests:
+    name: Run pytest tests
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      - name: Setup Python environment
+        uses: actions/setup-python@v5
+        with:
+          python-version: "3.x"
+          cache: 'pip'
+          cache-dependency-path: setup.py
+
+      - name: Install dependencies
+        run: pip install -e .[dev]
+
+      - name: Run Integration Tests
+        run: pytest tests/glassflow/integration_tests/
+        env:
+          PIPELINE_ID: ${{ secrets.PIPELINE_ID }}
+          PIPELINE_ACCESS_TOKEN: ${{ secrets.PIPELINE_ACCESS_TOKEN }}
```

Comment:
It doesn't seem right. 
At least, let's rename it into something more concrete, like `INTEGRATION_TESTS_PIPELINE_ID`.

Reply:
Base URL also needs to be configured - we shouldn't run integration tests on production environment.
At least, it could be `staging`.

Reply:
But anyway, conceptually we need to do integration tests differently.

Reply:
Good points! We can rename the secrets.

As for the environment, agree, we should test on a staging environment.